src/main/java/frc/robot/config/DynamicRobotConfig.java

```diff
@@ -0,0 +1,67 @@
+package frc.robot.config;
+
+import edu.wpi.first.wpilibj.Preferences;
+import frc.robot.Robot;
+import java.lang.reflect.Field;
+import java.util.logging.Logger;
+
+/**
+ * The DynamicRobotConfig class keeps track of periodically changing constants on the robot
+ *
+ * <p>This should only have offsets that would change with a mechanical change. There should be no
+ * IDs, Conversions, or setpoint information in this config.
+ */
+public class DynamicRobotConfig {
+  private static class ConfigVariables {
+    public static double frontLeftOffset = 0;
+    public static double frontRightOffset = 0;
+    public static double backLeftOffset = 2;
+    public static double backRightOffset = 0;
+  }
+
+  private static Logger logger = Logger.getLogger(DynamicRobotConfig.class.getName());
+
+  private DynamicRobotConfig() {
+    // This code interacts with ConfigVariables using reflection, essentially using it as a map.
+    // First, the code gets a reference to every variable from the static class, referred to as a
+    // "Field"
+    // It then iterates through this list of references
+    for (Field variable : ConfigVariables.class.getFields()) {
+      // For each field, its name is stored as a string, so it can be used as a preference key
+      String key = variable.getName();
+      try {
+        // If the preference does not exist, or we are bypassing to competition robot defaults, log
+        // an alert and open network access for that preference
+        if (!Preferences.containsKey(key) || Robot.isCompetition) {
+          raiseWarning("Using competition robot default for " + key);
+          Preferences.initDouble(key, variable.getDouble(ConfigVariables.class));
```

Comment:
Consider how we will support other Preference data types.

Reply:
I could implement functionality for that, but as it stands can we foresee a use case for anything other than doubles?

Reply:
Booleans and Integers seem likely. They can be future iterations, but we won't want to have to re-write everything to get that support.